python/dgl/graphbolt/impl/fused_csc_sampling_graph.py

```diff
@@ -426,6 +454,7 @@ def _convert_to_sampled_subgraph(
             # The sampled graph is already a homogeneous graph.
             sampled_csc = CSCFormatBase(indptr=indptr, indices=indices)
         else:
+            # UVA sampling requires us to move node_type_offset to GPU.
             self.node_type_offset = self.node_type_offset.to(column.device)
```

Comment:
Why not we only change this line to `node_type_offset=self.node_type_offset.to(column.device)`, and still use `self.node_type_offset` in other places that prefer CPU?

Reply:
Because when we move the graph to GPU, `self.node_type_offset` is moved as well. Then, there is synchronizations at every access to `self.node_type_offset`. If we keep moving to the GPU here at every iteration, then we pay the copy cost every iteration.

Reply:
So we basically need to have two versions, one on the GPU, one on the CPU.

Reply:
So that we use the approproate one depending on the operation.

Reply:
Make sense.